plugins/chat/app/models/chat/channel.rb

```diff
@@ -164,6 +164,27 @@ def latest_not_deleted_message_id(anchor_message_id: nil)
       SQL
     end
 
+    def mark_all_threads_as_read(user: nil)
```

Comment:
Should we check if the channel still has threading enabled before executing the query? Since this happens in a background job, there is a chance that the channel threading might have been disabled before the job is executed.

Reply:
Yeah I will add this, wasn't sure whether it was necessary but it is a good point that threading could get turned off inbetween this happening.